src/DynamoCore/DSEngine/NodeToCode.cs

```diff
@@ -1,16 +1,692 @@
 ﻿using System.Collections.Generic;
-
+using System.Linq;
 using Dynamo.Models;
+using Dynamo.Nodes;
+using System;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.DSASM;
+using Dynamo.Utilities;
 
 namespace Dynamo.DSEngine
 {
-    public static class NodeToCodeUtils
+    /// <summary>
+    /// The result of converting nodes to code. As when a node is converted to 
+    /// code, its inputs and outputs may be renamed to avoid confliction, the
+    /// result contains maps for inputs/new-inputs and outputs/new-outputs.
+    /// </summary>
+    public class NodeToCodeResult
+    {
+        /// <summary>
+        /// AST nodes that compiled from NodeModel.
+        /// </summary>
+        public IEnumerable<AssociativeNode> AstNodes { get; private set; }
+
+        /// <summary>
+        /// The map between original input name and new name.
+        /// </summary>
+        public Dictionary<string, string> InputMap { get; private set; }
```

Comment:
Just need some clarifications, if I have two `+` nodes (which both have `x` and `y` input ports), wouldn't there be two entries in this dictionary with the same key `x`?